edx_event_bus_redis/internal/tests/test_consumer.py

```diff
@@ -158,31 +159,40 @@ def raise_exception():
             # Final "call" just serves to stop the loop
             self.event_consumer._shut_down  # pylint: disable=protected-access
         ]
+        mock_read_value = [(b'1', self.normal_message.to_binary_dict())]
+        mock_pending_value = None
+        side_effect_method = 'emit_signals_from_message'
+        if is_pending:
+            mock_pending_value = [{'message_id': b'some-id'}]
 
         with patch.object(
-                self.event_consumer, 'emit_signals_from_message',
-                side_effect=side_effects(mock_emit_side_effects),
-        ) as mock_emit:
+            self.event_consumer, side_effect_method,
+            side_effect=side_effects(mock_emit_side_effects),
+        ) as mock_method:
             mock_consumer = MagicMock(
-                **{'read.return_value': [(b'1', self.normal_message.to_binary_dict())], 'pending.return_value':
-                    pending_return_value, '__getitem__.return_value': (b'1', self.normal_message.to_binary_dict())},
+                **{
+                    'read.return_value': mock_read_value,
+                    'pending.return_value': mock_pending_value,
+                    '__getitem__.return_value': (b'1', self.normal_message.to_binary_dict())
+                },
                 autospec=True
             )
             self.event_consumer.db = Mock()
             self.event_consumer.consumer = mock_consumer
             self.event_consumer.consume_indefinitely()
 
         # Check that each of the mocked out methods got called as expected.
-        if pending_return_value:
-            mock_consumer.pending.assert_called_with(count=1, consumer='test_group_id.c1')
+        mock_consumer.pending.assert_called_with(count=1, consumer='test_group_id.c1')
+        if not is_pending:
+            mock_consumer.read.assert_called()
```

Comment:
We should also check that `read` was not called if is_pending is true, since that would mean we had skipped a pending message.